prefect_aws/client_wait.py

```diff
@@ -0,0 +1,82 @@
+"""Task for waiting on a long-running AWS job"""
+
+import importlib.resources as pkg_resources
+import json
+from functools import partial
+from typing import Any, Dict, Optional
+
+import boto3
+from anyio import to_thread
+from botocore.exceptions import WaiterError
+from botocore.waiter import WaiterModel, create_waiter_with_client
+from prefect import get_run_logger, task
+
+from prefect_aws import waiters
+from prefect_aws.credentials import AwsCredentials
+
+
+@task
+async def client_waiter(
+    client: str,
+    waiter_name: str,
+    waiter_definition: Dict[str, Any],
+    aws_credentials: AwsCredentials,
+    **waiter_kwargs: Optional[Dict[str, Any]],
+):
+    """
+    Uses the underlying boto3 waiter functionality.
+
+    Args:
+        client: The AWS client on which to wait (e.g., 'batch', 'ec2', etc).
+        waiter_name: The name of the waiter to instantiate. Can be a boto-supported
+            waiter or one of prefect's custom waiters. Currently, Prefect offers three additional
+            waiters for AWS Batch: `"JobExists"` waits for a job to be instantiated, `"JobRunning"`
+            waits for a job to start running, and `"JobComplete"` waits for a job to finish. You can
+            find the definitions for all prefect-defined waiters [here](
+            https://github.com/PrefectHQ/prefect-aws/tree/master/prefect_aws/waiters/).
+            You may also use a custom waiter name, if you supply an accompanying waiter definition
+            dict.
+        waiter_definition: A valid custom waiter model, as a dict. Note that if
+            you supply a custom definition, it is assumed that the provided 'waiter_name' is
+            contained within the waiter definition dict.
+        credentials: your AWS credentials passed from an upstream
+            Secret task; this Secret must be a JSON string
+            with two keys: `ACCESS_KEY` and `SECRET_ACCESS_KEY` which will be
+            passed directly to `boto3`.  If not provided here or in context, `boto3`
+            will fall back on standard AWS rules for authentication.
+        waiter_kwargs: Arguments to pass to the `waiter.wait(...)` method. Will
+            depend upon the specific waiter being called.
+
+    Examples:
+        TODO:
```

Comment:
Try to do a full integration test flow, ensuring this task actually works, and then replace your credentials with makeup credentials